code/game/gamemodes/vampire/vampire_powers.dm

```diff
@@ -269,16 +269,24 @@
 			var/mob/living/carbon/human/H = C
 			if(H.check_ear_prot() >= HEARING_PROTECTION_TOTAL)
 				continue
+			if(H.check_ear_prot() >= HEARING_PROTECTION_MINOR)
+				to_chat(C, "<span class='warning'><font size='3'><b>You hear a dull shriek and instinctively cover your ears!</font></b></span>")
+				C.Confused(10)
+				C.Stuttering(10)
+				C.Jitter(50)
+				continue
 		if(!affects(C))
 			continue
```

Comment:
you want the `if(!affects(C))` check to come before this confuse bit. if not it will confuse other vampires, which is probably not what you want.

Reply:
Done, thanks !